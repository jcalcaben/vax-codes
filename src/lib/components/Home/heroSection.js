import React from "react";
import { StaticImage } from "gatsby-plugin-image";

import {
  root,
  callToAction,
  heading,
  summary,
  image,
  attribution,
  attributionLink,
} from "./heroSection.module.css";

const HeroSection = (props) => {
  const { headingText, summaryText } = props;

  return (
    <div className={root}>
      <section className={callToAction}>
        <h1 className={heading}>{headingText}</h1>
        <p className={summary}>{summaryText}</p>
      </section>
      <div className={image}>
        <StaticImage
          src="./images/homeHero.jpg"
          alt="Man in Gray Long Sleeve Shirt Holding Baby in White Onesie"
        />
        <div>
          <span className={attribution}>Public Domain: </span>
          <a
            className={attributionLink}
            href="https://www.pexels.com/photo/man-in-gray-long-sleeve-shirt-holding-baby-in-white-onesie-6973197/"
          >
            Mikhail Nilov
          </a>
        </div>
      </div>
    </div>
  );
};

export default HeroSection;
