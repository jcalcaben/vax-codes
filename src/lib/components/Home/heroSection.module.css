.root {
	composes: grid from global;
	composes: grid-cols-2 from global;
	composes: my-10 from global;
}

.attribution {

}

.attributionLink {

}

.callToAction {

}

.heading {
	composes: text-2xl from global;
	composes: my-6 from global;
	composes: font-bold from global;
}

.image {

}

.summary {
	composes: text-lg from global;
}