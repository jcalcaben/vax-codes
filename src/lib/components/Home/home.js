import React from "react";

import HeroSection from "./heroSection";

const Home = (props) => {
  const { headingText, summaryText } = props;

  const heroSectionProps = { headingText, summaryText };
  return (
    <>
      <HeroSection {...heroSectionProps} />
    </>
  );
};

export default Home;
