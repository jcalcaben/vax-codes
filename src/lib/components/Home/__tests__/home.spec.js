import React from "react";
import Home from "../home";

import TestRenderer from "react-test-renderer";

jest.mock("gatsby-plugin-image");

describe("Home component", () => {
  test("renders the correct basic shape", () => {
    const homeProps = { headingText: "Heading", summaryText: "Lorem ipsum" };

    const tree = TestRenderer.create(<Home {...homeProps} />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
