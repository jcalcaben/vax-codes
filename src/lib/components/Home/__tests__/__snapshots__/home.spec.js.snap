// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders the correct basic shape 1`] = `
<div
  className="root"
>
  <section
    className="callToAction"
  >
    <h1
      className="heading"
    >
      Heading
    </h1>
    <p
      className="summary"
    >
      Lorem ipsum
    </p>
  </section>
  <div
    className="image"
  >
    <img
      alt="Man in Gray Long Sleeve Shirt Holding Baby in White Onesie"
      src="./images/homeHero.jpg"
    />
    <div>
      <span
        className="attribution"
      >
        Public Domain: 
      </span>
      <a
        className="attributionLink"
        href="https://www.pexels.com/photo/man-in-gray-long-sleeve-shirt-holding-baby-in-white-onesie-6973197/"
      >
        Mikhail Nilov
      </a>
    </div>
  </div>
</div>
`;
